# Description

##### The goal of this project is to apply data science techniques to discover information in two distinct datasets. The first dataset describes various speech signal processing algorithms applied to the speech recordings of Parkinson's Disease (PD) patients to extract clinically useful information for PD assessment. The second dataset predicts forest cover type from cartographic variables only. It includes four wilderness areas located in the Roosevelt National Forest of northern Colorado. These areas represent forests with minimal human-caused disturbances, so that existing forest cover types are more a result of ecological processes rather than forest management practices.

#### Parkinson's Disease Dataset

In [3]:
import pandas as pd

data1 = pd.read_csv('pd_speech_features.csv', sep=',')

data1.shape

(756, 755)

In [4]:
cat_vars = data1.select_dtypes(include='category')
cat_vars.columns.size

0

In [5]:
num_vars = data1.select_dtypes(include='number')
num_vars.columns.size

755

In [6]:
null_vars = {}
for var in data1:
    nas = data1[var].isna().sum()
    if nas > 0:
        null_vars[var] = nas
print(len(null_vars))

0


In [7]:
data1.head(6)

id  gender      PPE      DFA     RPDE  numPulses  numPeriodsPulses  \
0   0       1  0.85247  0.71826  0.57227        240               239   
1   0       1  0.76686  0.69481  0.53966        234               233   
2   0       1  0.85083  0.67604  0.58982        232               231   
3   1       0  0.41121  0.79672  0.59257        178               177   
4   1       0  0.32790  0.79782  0.53028        236               235   
5   1       0  0.50780  0.78744  0.65451        226               221   

   meanPeriodPulses  stdDevPeriodPulses  locPctJitter  ...  \
0          0.008064            0.000087       0.00218  ...   
1          0.008258            0.000073       0.00195  ...   
2          0.008340            0.000060       0.00176  ...   
3          0.010858            0.000183       0.00419  ...   
4          0.008162            0.002669       0.00535  ...   
5          0.007631            0.002696       0.00783  ...   

   tqwt_kurtosisValue_dec_28  tqwt_kurtosisValue_dec_29  \
0                     1.5620                     2.6445   
1                     1.5589                     3.6107   
2                     1.5643                     2.3308   
3                     3.7805                     3.5664   
4                     6.1727                     5.8416   
5                     4.8025                     5.0734   

   tqwt_kurtosisValue_dec_30  tqwt_kurtosisValue_dec_31  \
0                     3.8686                     4.2105   
1                    23.5155                    14.1962   
2                     9.4959                    10.7458   
3                     5.2558                    14.0403   
4                     6.0805                     5.7621   
5                     7.0166                     5.9966   

   tqwt_kurtosisValue_dec_32  tqwt_kurtosisValue_dec_33  \
0                     5.1221                     4.4625   
1                    11.0261                     9.5082   
2                    11.0177                     4.8066   
3                     4.2235                     4.6857   
4                     7.7817                    11.6891   
5                     5.2065                     7.4246   

   tqwt_kurtosisValue_dec_34  tqwt_kurtosisValue_dec_35  \
0                     2.6202                     3.0004   
1                     6.5245                     6.3431   
2                     2.9199                     3.1495   
3                     4.8460                     6.2650   
4                     8.2103                     5.0559   
5                     3.4153                     3.5046   

   tqwt_kurtosisValue_dec_36  class  
0                    18.9405      1  
1                    45.1780      1  
2                     4.7666      1  
3                     4.0603      1  
4                     6.1164      1  
5                     3.2250      1  

[6 rows x 755 columns]

##### The dataset has 756 records and 755 attributes, all numeric, and does not have any null value.

#### Covertype Dataset

# Preprocessing

### Reducing Sample Size

#### Parkinson's Disease Dataset

##### This dataset describes three measurements for each patient, so we apply the mean of those samples to have one sample per patient.

In [9]:
data1 = data1.groupby('id').mean().reset_index()

data1.shape

(252, 755)

In [10]:
data1.head(6)

id  gender       PPE       DFA      RPDE   numPulses  numPeriodsPulses  \
0   0     1.0  0.823387  0.696370  0.567250  235.333333        234.333333   
1   1     0.0  0.415637  0.793993  0.592453  213.333333        211.000000   
2   2     1.0  0.801973  0.619967  0.520563  319.333333        318.333333   
3   3     0.0  0.828707  0.626097  0.537183  493.000000        492.000000   
4   4     0.0  0.831287  0.779397  0.726717  362.666667        361.666667   
5   5     1.0  0.822520  0.622083  0.357660  284.000000        283.000000   

   meanPeriodPulses  stdDevPeriodPulses  locPctJitter  ...  \
0          0.008220            0.000073      0.001963  ...   
1          0.008884            0.001849      0.005790  ...   
2          0.006041            0.000104      0.002217  ...   
3          0.003913            0.000042      0.000757  ...   
4          0.005622            0.002023      0.003593  ...   
5          0.006815            0.000046      0.000940  ...   

   tqwt_kurtosisValue_dec_28  tqwt_kurtosisValue_dec_29  \
0                   1.561733                   2.862000   
1                   4.918567                   4.827133   
2                  41.129400                  31.201933   
3                   1.677633                   1.908400   
4                   4.104600                   4.285233   
5                  25.482500                  29.795367   

   tqwt_kurtosisValue_dec_30  tqwt_kurtosisValue_dec_31  \
0                  12.293333                   9.717500   
1                   6.117633                   8.599667   
2                  14.584467                   5.446800   
3                   2.842167                   3.493867   
4                   2.953200                   2.799933   
5                  26.472767                  43.983700   

   tqwt_kurtosisValue_dec_32  tqwt_kurtosisValue_dec_33  \
0                   9.055300                   6.259100   
1                   5.737233                   7.933133   
2                   3.462000                   4.772067   
3                   3.282433                   3.085267   
4                   2.645100                   2.811367   
5                  54.324967                  49.879867   

   tqwt_kurtosisValue_dec_34  tqwt_kurtosisValue_dec_35  \
0                   4.021533                   4.164333   
1                   5.490533                   4.941833   
2                   9.176633                  11.848100   
3                   3.184433                   4.032933   
4                   7.268333                  13.338833   
5                  43.996667                  37.834733   

   tqwt_kurtosisValue_dec_36  class  
0                  22.961700    1.0  
1                   4.467233    1.0  
2                   5.552367    1.0  
3                  22.773633    1.0  
4                  63.766900    1.0  
5                  73.894367    1.0  

[6 rows x 755 columns]

### Feature Selection

# Classification